Peaky.Tests/TestDiscoveryTests.cs

```diff
@@ -545,198 +553,22 @@ public void when_a_test_does_not_accept_input_parameters_then_queryParameters_is
-
-    public class OrangeTests : IHaveTags
-    {
-        public string[] Tags => new[]
-        {
-            "orange", "fruit"
-        };
-
-        public dynamic tangerine() => "Oooh!";
-    }
-
-    public class InternalOnlyTests : IApplyToEnvironment
-    {
-        public dynamic internal_only_test() => "success!";
-
-        public bool AppliesToEnvironment(string environment) => !environment.Equals("production", StringComparison.OrdinalIgnoreCase);
-    }
-
-    public class WidgetApiTests : IApplyToApplication
-    {
-        public dynamic widgetapi_only_test() => "success!";
-
-        public bool AppliesToApplication(string application) => application.Equals("widgetapi", StringComparison.OrdinalIgnoreCase);
-    }
-
-    public class CollisionTest1 : IPeakyTest
-    {
-        public dynamic name_collision()
-        {
-            return GetType().Name;
-        }
-    }
-
-    public class CollisionTest2 : IPeakyTest
-    {
-        public dynamic name_collision()
-        {
-            return GetType().Name;
-        }
-    }
-
-    public class TestDiscoveryResponse
-    {
-        public Test[] Tests { get; set; }
-
-        public class Test
+        [Fact]
+        public void when_a_test_exposes_parametrized_test_cases_then_the_input_parameters_are_recorded()
         {
-            public string Application { get; set; }
-            public string Environment { get; set; }
-            public string Url { get; set; }
-            public Parameter[] Parameters { get; set; }
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/parametrized/").Result;
 
-            public class Parameter
-            {
-                public string Name { get; set; }
-                public object DefaultValue { get; set; }
-            }
+            response.ShouldSucceed();
+            var content = JsonConvert.DeserializeObject<TestDiscoveryResponse>(response.Content.ReadAsStringAsync().Result);
+            
+            content.Tests.Should().Contain(t => t.Url.ToString().EndsWith("I_do_stuff/?extectedResult=true&testCaseId=case1", StringComparison.OrdinalIgnoreCase));
+            content.Tests.Should().Contain(t => t.Url.ToString().EndsWith("I_do_stuff/?extectedResult=false&testCaseId=case2", StringComparison.OrdinalIgnoreCase));
+            content.Tests.Should().Contain(t => t.Url.ToString().EndsWith("I_do_stuff/?extectedResult=true&testCaseId=case3", StringComparison.OrdinalIgnoreCase));
+            content.Tests.Should().Contain(t => t.Url.ToString().EndsWith("I_do_stuff/?extectedResult=false&testCaseId=case4", StringComparison.OrdinalIgnoreCase));
+
+            content.Tests.Should().Contain(t => t.Url.ToString().EndsWith("I_do_stuff_and_return/?extectedResult=true&testCaseId=case1", StringComparison.OrdinalIgnoreCase));
+            content.Tests.Should().Contain(t => t.Url.ToString().EndsWith("I_do_stuff_and_return/?extectedResult=false&testCaseId=case2", StringComparison.OrdinalIgnoreCase));
```

Comment:
Do we need this many assertions to know the scenario is working? Are the different assertions possibly different (and likely enough to fail independently) that they deserve to be separate tests?